ci: new github workflow for license analysis with FOSSA

clementnero opened the pull request:
## Description

Related to https://github.com/camunda/team-infrastructure/issues/751.

Add a new workflow to analyze licenses using FOSSA.

This workflow uploads only a dependency graph to FOSSA, the source code itself is not uploaded; The file-level scan of the dependencies is performed by the FOSSA platform asynchronously, making/keeping this workflow fast (execution time).

It currently analyzes licenses without enforcing any blocking checks and targets the main branch. Tag analysis will be added later, along with PR checks.

Dependencies within the `import`, `provided`, `system` and `test` (Maven) scopes are ignored.

The following projects are ignored (to be check/confirm during review):
- `./.github/optimize/scripts/healthStatusReport`
- `./c8run/e2e_tests`
- `./qa/core-application-e2e-test-suite`
- `./tasklist/qa/backup-restore-tests`

The following Maven modules are also ignored (to be check/confirm during review):
- .`/operate`
    - `io.camunda:operate-qa`
    - `io.camunda:operate-qa-backup-restore-tests`
    - `io.camunda:operate-qa-data-generator`
    - `io.camunda:operate-qa-it-tests`
    - `io.camunda:operate-qa-migration-performance-test`
    - `io.camunda:operate-qa-migration-tests`
    - `io.camunda:operate-qa-migration-tests-parent`
    - `io.camunda:operate-qa-migration-tests-test-fixture-110`
    - `io.camunda:operate-qa-migration-tests-test-fixture-120`
    - `io.camunda:operate-qa-migration-tests-test-fixture-130`
    - `io.camunda:operate-qa-migration-tests-test-fixture-800`
    - `io.camunda:operate-qa-migration-tests-test-fixture-810`
    - `io.camunda:operate-qa-migration-tests-test-fixture-820`
    - `io.camunda:operate-qa-migration-tests-test-fixture-830`
    - `io.camunda:operate-qa-migration-tests-test-fixture-840`
    - `io.camunda:operate-qa-migration-tests-test-fixture-850`
    - `io.camunda:operate-qa-query-performance-tests`
    - `io.camunda:operate-qa-util`
- `./qa`
    - `io.camunda:camunda-qa`
    - `io.camunda:camunda-qa-integration-tests`
    - `io.camunda:camunda-qa-util`
- `./tasklist`
    - `io.camunda:tasklist-qa`
    - `io.camunda:tasklist-qa-backup-restore-tests`
    - `io.camunda:tasklist-qa-migration-tests`
    - `io.camunda:tasklist-qa-migration-tests-parent`
    - `io.camunda:tasklist-qa-migration-tests-test-fixture-810`
    - `io.camunda:tasklist-qa-migration-tests-test-fixture-820`
    - `io.camunda:tasklist-qa-migration-tests-test-fixture-830`
    - `io.camunda:tasklist-qa-migration-tests-test-fixture-840`
    - `io.camunda:tasklist-qa-migration-tests-test-fixture-850`
    - `io.camunda:tasklist-test-coverage`
- `./testing`
    - `io.camunda:camunda-process-test-spring`
- `./zeebe`
    - `io.camunda:zeebe-qa`
    - `io.camunda:zeebe-qa-integration-tests`
    - `io.camunda:zeebe-qa-update-tests`
    - `io.camunda:zeebe-qa-util`
    - `io.camunda:zeebe-protocol-test-util`
    - `io.camunda:zeebe-test-util`

Test workflow run: https://github.com/camunda/camunda/actions/runs/14086556963

Before merging:
- [ ] check that https://github.com/camunda/infra-global-github-actions/pull/373 is merged (github composite actions)
    - [ ]  replace the @main ref by a commit hash 

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
  - [ ] enable backports [when recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #

kellervater commented on .github/workflows/check-licenses.yml:
```diff
@@ -0,0 +1,77 @@
+          path: ./
+          maven-config: |
+            -DskipQaBuild=true
+        # To be uncomment when Optimize is fully sperated
```
```suggestion
        # To be uncommented when Optimize is fully seperated
```

kellervater reviewed (changes requested):
Without much knowledge about what needs to be excluded here (assuming everything what doesn't end up in our deliverables), I'd say :rocket: LGTM.

One remark though from the [github-advanced-security](https://github.com/apps/github-advanced-security) bot about the workflow permissions: I think @cmur2 enforced that we need to set permissions on monorepo workflows to restrict the standard github token.

cmur2 commented:
> enforced that we need to set permissions on monorepo workflows to restrict the standard github token.

It is not enforced but as the bot suggested please follow the [best practices](https://github.com/camunda/camunda/wiki/CI-&-Automation#permissions-of-github_token).

kellervater reviewed (approved):
lgtm :rocket: 

Not sure how the `Workflow does not contain permissions` reported by advanced security got fixed now (due to the squash commit), but apparently it is :smile:

clementnero commented:
I added

> lgtm 🚀
> 
> Not sure how the `Workflow does not contain permissions` reported by advanced security got fixed now (due to the squash commit), but apparently it is 😄

Fixed by https://github.com/camunda/camunda/pull/30246/files#diff-551809b48b3c0309a29e798b46279f933c6d4446154fe3e3b30603076c1be547R11

clementnero commented:
/backport

backport-action commented:
Successfully created backport PR for `stable/8.4`:
- #30755

backport-action commented:
Successfully created backport PR for `stable/8.5`:
- #30756

backport-action commented:
Created backport PR for `stable/8.6`:
- #30757 with remaining conflicts!

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin backport-30246-to-stable/8.6
git worktree add --checkout .worktree/backport-30246-to-stable/8.6 backport-30246-to-stable/8.6
cd .worktree/backport-30246-to-stable/8.6
git reset --hard HEAD^
git cherry-pick -x 9b683e641971273a10301fafcfae93de95c6f600
git push --force-with-lease
```

backport-action commented:
Successfully created backport PR for `stable/optimize-8.7`:
- #30760
